src/NuGet.Clients/NuGet.VisualStudio.Internal.Contracts/NuGetProjectKind.cs

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+namespace NuGet.VisualStudio.Internal.Contracts
+{
+    public enum NuGetProjectKind
+    {
+        Unknown,
+        Classic,
+        ProjectK,
```

Comment:
I think ProjectK is xproj...so it's a codepath that won't get hit in VS 2017 and later, so no need to special case it :)